extension/httpfs/create_secret_functions.cpp

```diff
@@ -69,14 +85,74 @@ unique_ptr<BaseSecret> CreateS3SecretFunctions::CreateSecretFunctionInternal(Cli
 			secret->secret_map["url_compatibility_mode"] = Value::BOOLEAN(named_param.second.GetValue<bool>());
 		} else if (lower_name == "account_id") {
 			continue; // handled already
+		} else if (lower_name == "refresh") {
+			refresh = named_param.second.GetValue<bool>();
+			secret->secret_map["refresh"] = Value::BOOLEAN(refresh);
+			child_list_t<Value> struct_fields;
+			for (const auto &named_param : input.options) {
+				auto lower_name = StringUtil::Lower(named_param.first);
+				struct_fields.push_back({lower_name, named_param.second});
+			}
+			secret->secret_map["refresh_info"] = Value::STRUCT(struct_fields);
+		} else if (lower_name == "refresh_info") {
+			if (refresh) {
+				throw InvalidInputException("Can not set `refresh` and `refresh_info` at the same time");
+			}
```

Comment:
should this be there also in the previous block, to check that `refresh` is not set after `refresh_info`?

Reply:
correct